packages/babel-plugin-transform-merge-sibling-variables/src/index.js

```diff
@@ -90,10 +90,15 @@ module.exports = function({ types: t }) {
               return;
             }
 
-            init.node.declarations = node.declarations.concat(
+            const declarations = node.declarations.concat(
               init.node.declarations
             );
+
+            // temporary workaround to forces babel recalculate scope,
+            // references and binding until babel/babel#4818 resolved
             path.remove();
+            init.remove();
+            next.node.init = t.variableDeclaration("var", declarations);
```

Comment:
```js
next.replaceWith(t.variableDeclaration("var", declarations))
```

Reply:
Can you also fix this one? Use babel path API instead of mutating node.

Reply:
Tried & failed with this way, the output eliminated the for-loop

Reply:
Oops. It's 

```js
init.replaceWith(t.variableDeclaration("var", declarations);
```

and `init.remove()` should be removed.

Reply:
Done, thanks for correcting